src/app/src/main/kotlin/org/vaccineimpact/api/app/controllers/OneTimeLinkController.kt

```diff
@@ -28,7 +32,43 @@ class OneTimeLinkController(
         val token = context.params(":token")
         val claims = verifyToken(token, repo)
         val link = OneTimeLink.parseClaims(claims)
-        return link.perform(controllers, context, repos)
+        val redirectUrl = link.queryParams["redirectUrl"]
```

Comment:
Maybe make sure here that `redirectUrl` is always `null` or something, so that later you can use `redirectUrl == null` and smart casting.